### PR TITLE
prevent celery command autoscale misconfig

### LINKED ISSUE #35332: WORKER_AUTOSCALE should not allow max workers < min workers
### Apache Airflow version

Other Airflow 2 version (please specify below)

### What happened

[This was reported for Airflow 2.5.3](https://apache-airflow.slack.com/archives/CCQ7EGB1P/p1698124263201309), but I suspect the behavior is still the same as I don't see any related issues/PRs.

The [WORKER_CONCURRENCY configuration is rather counterintuitive](https://github.com/apache/airflow/pull/4609); the configuration is specified by `max workers, min workers`. This can lead users to specify a maximum number of workers that is less than the minimum number of workers. I'm not entirely sure what happens when this is specified, but I don't believe it behaves normally.

### What you think should happen instead

An exception should be raised if a user attempts to specify a maximum number of workers less than the minimum number of workers.

### How to reproduce

Set `AIRFLOW__CELERY__WORKER_AUTOSCALE=1, 3`, and I suspect only 1 node will ever exist.

https://airflow.apache.org/docs/apache-airflow-providers-celery/stable/configurations-ref.html#worker-autoscale

### Operating System

n/a

### Versions of Apache Airflow Providers

_No response_

### Deployment

Other

### Deployment details

_No response_

### Anything else

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by Bowrna**
closes: #35332 
related: #35332 

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/cli/commands/celery_command.py [hunk 1]
```diff
@@ -131,6 +131,10 @@ def worker(args):
 
     if autoscale is None and conf.has_option("celery", "worker_autoscale"):
         autoscale = conf.get("celery", "worker_autoscale")
+        max_worker, min_worker = [int(x) for x in test.split("-")]
+        if min_worker > max_worker:
+            raise SystemExit("The option for worker autoscale should be max_concurrency,min_concurrency")
+
```
Comment (dirrao) [inline airflow/cli/commands/celery_command.py]:
```suggestion
    max_worker, min_worker = [int(x) for x in autoscale.split("-")]
    if min_worker > max_worker:
          raise SystemExit("The option for worker autoscale should be max_concurrency,min_concurrency")
```
This validation should be done for default configuration as well.

**Review by dirrao — CHANGES_REQUESTED** [review]
Static checks are failing. I would request you to use pre commit hooks or breeze CLI static checks.
https://github.com/apache/airflow/blob/main/STATIC_CODE_CHECKS.rst#:~:text=available%20bash%20scripts.-,Pre%2Dcommit%20hooks,on%20and%20make%20them%20fast.

**Review by dirrao — CHANGES_REQUESTED** [review]
Existing test cases are failing. Can you update the existing ones and add the new test case?

**Comment by Bowrna** [comment]
hi @dirrao yes the static checks are expected to fail. I will need response for this comment https://github.com/apache/airflow/issues/35332#issuecomment-1876874115 to proceed further with this.

**Comment by Bowrna** [comment]
i have fixed some parts in documentation alone here in this PR. The actual issue is not an issue and therefore I just enhanced the docs alone

**Review by potiuk — APPROVED** [review]

**Pull request merged**